From 09c6702f74c8e4c2af700ebc8006c40083ce5b72 Mon Sep 17 00:00:00 2001
From: Ayoub Zaki <ayoub.zaki@embetrix.com>
Date: Sun, 26 Apr 2026 17:50:00 +0200
Subject: [PATCH] params: raise charp parameter length limit to 4096

param_set_charp() caps charp module parameters at 1024 bytes.
dm-mod.create with dm-verity root_hash_sig_hex passes a PKCS#7
signature as hex, which exceeds this limit.  Raise to 4096.

Upstream-Status: Inappropriate

Signed-off-by: Ayoub Zaki <ayoub.zaki@embetrix.com>
---
 kernel/params.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/params.c b/kernel/params.c
index b96cfd693c99..8425f0bf36c6 100644
--- a/kernel/params.c
+++ b/kernel/params.c
@@ -261,7 +261,7 @@ EXPORT_SYMBOL_GPL(param_set_uint_minmax);

 int param_set_charp(const char *val, const struct kernel_param *kp)
 {
-	size_t len, maxlen = 1024;
+	size_t len, maxlen = 4096;

 	len = strnlen(val, maxlen + 1);
 	if (len == maxlen + 1) {
